packages/backend/src/open_payments/payment/outgoing/lifecycle.ts

```diff
@@ -42,95 +38,76 @@ export async function handleSending(
   validateAssets(deps, payment, receiver)
 
   // Due to SENDING→SENDING retries, the quote's amount parameters may need adjusting.
-  const newMaxSourceAmount = payment.debitAmount.value - amountSent
-
-  // This is only an approximation of the true amount delivered due to exchange rate variance. The true amount delivered is returned on stream response packets, but due to connection failures there isn't a reliable way to track that in sync with the amount sent.
-  // eslint-disable-next-line no-case-declarations
-  const amountDelivered = BigInt(
-    Math.ceil(+amountSent.toString() * payment.quote.minExchangeRate.valueOf())
+  const { maxReceiveAmount, maxDebitAmount } = getAdjustedAmounts(
+    deps,
+    payment,
+    receiver,
+    amountSent
   )
-  let newAmountToDeliver = payment.receiveAmount.value - amountDelivered
 
-  if (receiver.incomingAmount && receiver.receivedAmount) {
-    const maxAmountToDeliver =
-      receiver.incomingAmount.value - receiver.receivedAmount.value
-    if (maxAmountToDeliver < newAmountToDeliver) {
-      newAmountToDeliver = maxAmountToDeliver
-    }
-  }
-
-  if (newAmountToDeliver <= BigInt(0)) {
-    // Payment is already (unexpectedly) done. Maybe this is a retry and the previous attempt failed to save the state to Postgres. Or the incoming payment could have been paid by a totally different payment in the time since the quote.
+  if (maxReceiveAmount <= BigInt(0)) {
+    // Payment is already (unexpectedly) done.
+    // Maybe this is a retry and the previous attempt failed to save the state to Postgres. Or the incoming payment could have been paid by a totally different payment in the time since the quote.
     deps.logger.warn(
       {
-        newMaxSourceAmount,
-        newAmountToDeliver,
+        maxDebitAmount,
+        maxReceiveAmount,
         amountSent,
         receiver
       },
       'handleSending payment was already paid'
     )
     await handleCompleted(deps, payment)
     return
-  } else if (
-    newMaxSourceAmount <= BigInt(0) ||
-    newAmountToDeliver <= BigInt(0)
-  ) {
+  }
+
+  if (maxDebitAmount <= BigInt(0)) {
```

Comment:
Why is the second condition dropped?

Reply:
it wasn't necessary, since it was the same as the first `if` statement